homeassistant/components/sensor/onewire.py

```diff
@@ -19,21 +30,22 @@
 def setup_platform(hass, config, add_devices, discovery_info=None):
     """Setup the one wire Sensors."""
     base_dir = config.get('mount_dir', '/sys/bus/w1/devices/')
-    device_folders = glob(os.path.join(base_dir, '[10,22,28,3B,42]*'))
     sensor_ids = []
     device_files = []
-    for device_folder in device_folders:
-        sensor_ids.append(os.path.split(device_folder)[1])
-        if base_dir.startswith('/sys/bus/w1/devices'):
-            device_files.append(os.path.join(device_folder, 'w1_slave'))
-        else:
-            device_files.append(os.path.join(device_folder, 'temperature'))
+    for device_family in ('10', '22', '28', '3B', '42'):
```

Comment:
Don't know if that list change often. Perhaps would be worth to create a constant for that.